### PR TITLE
Implemented SQL "WHERE <field> IS [NOT] NULL"

### PR DESCRIPTION
**Pull request by paegun**
- implemented SQL "WHERE <field> IS [NOT] NULL".

### CODE REVIEW

File: src/riak_ql_ddl.erl [hunk 1]
```diff
@@ -315,9 +320,9 @@ is_filters_field_valid(Mod, {Op, Field, {RHS_type, RHS_Val}}, Acc1) ->
     case Mod:is_field_valid([Field]) of
         true  ->
             ExpectedType = Mod:get_field_type([Field]),
-            case is_compatible_type(ExpectedType, RHS_type, normalise(RHS_Val)) of
-                true  -> Acc2 = Acc1;
-                false -> Acc2 = [{incompatible_type, Field, ExpectedType, RHS_type} | Acc1]
+            Acc2 = case is_compatible_type(ExpectedType, RHS_type, normalise(RHS_Val)) of
```
Comment (hmmr) [inline src/riak_ql_ddl.erl:323]: It's always better to not let variables be bound within case branches. Well spotted.

File: src/riak_ql_parser.yrl [hunk 2]
```diff
@@ -191,6 +198,7 @@ Funcall -> Identifier left_paren FunArg         right_paren : make_funcall('$1',
 Funcall -> Identifier left_paren asterisk       right_paren : make_funcall('$1', ['$3']).
 Funcall -> Identifier left_paren FunArg FunArgN right_paren : make_funcall('$1', ['$3'] ++ '$4').
 
+NullCond -> Vals NullComp : make_expr('$1', '$2').
```
Comment (hmmr) [inline src/riak_ql_parser.yrl]: I suggest to rename "NullCond" to "NullPredicate" (as it is called in the SQL specs).

**Comment by hmmr** [comment]
+1

**Comment by hazen** [comment]
:+1: 2ebcc3d

**Pull request merged**